### PR TITLE
prov/efa: rename "rxr_op_entry" to "efa_rdm_ope"

### PR DESCRIPTION
**Pull request by wzamazon**
First step to retire the legacy term "rxr"

### CODE REVIEW

File: prov/efa/src/rdm/efa_rdm_ope.c [hunk 1]
```diff
@@ -158,21 +158,21 @@ void rxr_tx_entry_release(struct rxr_op_entry *tx_entry)
 		rxr_pkt_entry_release_tx(tx_entry->ep, pkt_entry);
 	}
 
-	if (tx_entry->rxr_flags & RXR_OP_ENTRY_QUEUED_RNR)
+	if (tx_entry->rxr_flags & efa_rdm_ope_QUEUED_RNR)
 		dlist_remove(&tx_entry->queued_rnr_entry);
 
-	if (tx_entry->rxr_flags & RXR_OP_ENTRY_QUEUED_CTRL)
+	if (tx_entry->rxr_flags & efa_rdm_ope_QUEUED_CTRL)
```
Comment (lrbison) [inline prov/efa/src/rdm/efa_rdm_ope.c:164]: fix case

Comment (lrbison) [reply prov/efa/src/rdm/efa_rdm_ope.c:164]: there are several _QUEUED_RNR and _QUEUED_CTRL throughout.

File: prov/efa/src/rdm/efa_rdm_ope.c [hunk 2]
```diff
@@ -1487,20 +1487,20 @@ int rxr_op_entry_post_remote_write(struct rxr_op_entry *op_entry)
...
 		EFA_WARN(FI_LOG_CQ, "Prepare to post read failed. err=%d\n", err);
 		return err;
 	}
 
-	err = rxr_op_entry_post_read(op_entry);
+	err = efa_rdm_ope_post_read(op_entry);
 	if (err == -FI_EAGAIN) {
 		dlist_insert_tail(&op_entry->queued_read_entry, &op_entry->ep->op_entry_queued_read_list);
-		op_entry->rxr_flags |= RXR_OP_ENTRY_QUEUED_READ;
+		op_entry->rxr_flags |= efa_rdm_ope_QUEUED_READ;
```
Comment (lrbison) [inline prov/efa/src/rdm/efa_rdm_ope.c]: and _QUEUED_READ

File: prov/efa/src/rdm/rxr_pkt_type_req.c [hunk 3]
```diff
@@ -1799,33 +1799,33 @@ void rxr_pkt_handle_longcts_rtw_sent(struct rxr_ep *ep,
...
 }
 
 /*
  *     handle_recv() functions
  */
 
 static
-struct rxr_op_entry *rxr_pkt_alloc_rtw_rx_entry(struct rxr_ep *ep,
+struct efa_rdm_ope *rxr_pkt_alloc_rtw_rx_entry(struct rxr_ep *ep,
 						struct rxr_pkt_entry *pkt_entry)
```
Comment (lrbison) [inline prov/efa/src/rdm/rxr_pkt_type_req.c:1826]: alignment... sigh.... are these worth tracking down?

File: prov/efa/src/rdm/efa_rdm_ope.c [hunk 4]
```diff
@@ -366,20 +366,20 @@ void rxr_tx_entry_set_runt_size(struct rxr_ep *ep, struct efa_rdm_ope *tx_entry)
  *
  * which is why this function is needed.
  *
- * @param[in]		op_entry		contains operation information
+ * @param[in]		ope		contains operation information
  * @param[in]		pkt_type		REQ packet type
  * @return			size of total data transfered by REQ packets
```
Comment (lrbison) [inline prov/efa/src/rdm/efa_rdm_ope.c:371]: probably lots of these too.  I'm inclined to ignore.

File: prov/efa/src/rdm/efa_rdm_ope.c [hunk 5]
```diff
@@ -924,51 +924,51 @@ void rxr_tx_entry_report_completion(struct efa_rdm_ope *tx_entry)
  * of all the packets that contains data.
  *
  * In both cases, the "all data has been send" event mark the end of the operation,
- * therefore this function will call rxr_tx/rx_op_entry_report_completion(), and
- * release the op_entry
+ * therefore this function will call rxr_tx/rx_ope_report_completion(), and
+ * release the ope
  *
- * @param[in]	op_entry	inforatminon of op entry that sends data
+ * @param[in]	ope	inforatminon of op entry that sends data
```
Comment (lrbison) [inline prov/efa/src/rdm/efa_rdm_ope.c:930]: unrelated: inforatminion is quote the mouthful.

File: prov/efa/src/rdm/efa_rdm_ope.h [hunk 6]
```diff
@@ -85,7 +85,7 @@ enum rxr_cuda_copy_method {
 	RXR_CUDA_COPY_LOCALREAD   /** device driven copy by using local RDMA read */
 };
 
-struct rxr_op_entry {
+struct efa_rdm_ope {
```
Comment (lrbison) [inline prov/efa/src/rdm/efa_rdm_ope.h:88]: I agree with the rename, but I think we are losing a little bit of good naming here.  Can we put a comment around this struct definition that "ope" is an "operation entry" just so it's documented somewhere?

**Review by lrbison — CHANGES_REQUESTED** [review]
All minor, with the exception of a few macro name changes which needs the case to be fixed.

File: prov/efa/src/rdm/efa_rdm_ope.c (hunk 1 above)
Comment (darrylabbate) [reply prov/efa/src/rdm/efa_rdm_ope.c:164]: Looks like a case of case-insensitive search and replace

Comment (wzamazon) [reply prov/efa/src/rdm/efa_rdm_ope.c]: Fixed

File: prov/efa/src/rdm/efa_rdm_ope.c (hunk 2 above)
Comment (wzamazon) [reply prov/efa/src/rdm/efa_rdm_ope.c]: Fixed

File: prov/efa/src/rdm/rxr_pkt_type_req.c (hunk 3 above)
Comment (wzamazon) [reply prov/efa/src/rdm/rxr_pkt_type_req.c:1826]: I would like to handle them via one patch when we are done with all the renaming.

File: prov/efa/src/rdm/efa_rdm_ope.h (hunk 6 above)
Comment (wzamazon) [reply prov/efa/src/rdm/efa_rdm_ope.h:102]: Added some doxgen string

**Comment by wzamazon** [comment]
@lrbison @darrylabbate 

I decided to focus on 1 change at a time, and redone the 1st patch.

**Review by lrbison — APPROVED** [review]
Thanks!

**Review by darrylabbate — APPROVED** [review]

**Comment by wzamazon** [comment]
pure EFA change, so ignore Intel CI

**Pull request merged**